### ISSUE TITLE
Add support for max runtime for workloads

### ISSUE THREAD

**Issue by kannon92** [kubernetes-sigs/kueue#405]
**What would you like to be added**:
It would be useful if ClusterQueues had a way to enforce walltime limit. 

This would enforce a walltime limit for any workload that is admitted into the ClusterQueue.   

**Why is this needed**:

In HPC schedulers, it is common to have multiple layers where users can request a job with a set walltime limit.  It is also common for admins to set up walltime limits.

A Batch user can submit a workload and should also be able to request a walltime limit for their job.  This can be useful because it allows a user to have greater control over how long their job will run for.  

An admin can design queues with various lengths for run time.  I've seen this in some HPC centers as a way to have dedicated queues for short lived jobs.  

One question I have is where should the user requested walltime limit live?   

**Completion requirements**:

This enhancement requires the following artifacts:

- [x] Design doc
- [x] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

**Comment by alculquicondor** [comment]
What do you need in addition to terminating the job once the deadline is hit?

Such field already exists in Job: https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-termination-and-cleanup
Although we don't utilize that information in kueue.

**Comment by kannon92** [comment]
Yea I saw that as I posted this.  In the Kueue design docs, it talks about having workloads as an abstraction for Jobs/CustomJobObjects.  That link is obviously specific to Jobs but I was wondering for other custom controllers.  But maybe we can call it solved for the JobController at the moment.  And other Job implementations would need their own deadline implemented.  

And are you okay with the addition to CQ?  Even if most users puts a deadline on their job, it may be good to have default limits on Queues for those that do not.
